index.js

```diff
@@ -71,12 +103,50 @@ function extendMemoryDB(MemoryDB) {
     };
   }
 
-  // Support exact key match filters only
-  function filter(snapshots, query) {
-    var mingoQuery = new Mingo.Query(query);
-    return snapshots.filter(function(snapshot) {
-      return snapshot.data && mingoQuery.test(snapshot.data);
-    });
+  function projectSnapshot(fields, snapshot) {
+    // Don't project when there's no projection
+    if (!fields) return snapshot;
+    // Don't project when called by ShareDB submit
+    if (fields.$submit) return snapshot;
+
+    var projectedData = {};
+    for (var key in snapshot.data) {
+      if (fields[key]) projectedData[key] = snapshot.data[key];
+    }
+
+    return {
+      id: snapshot.id,
+      v: snapshot.v,
+      type: snapshot.type,
+      m: snapshot.m,
+      o: snapshot.o,
+      data: projectedData
+    };
+  }
+
+  // Build a query object that mimics how the query would be executed if it were
+  // made against snapshots persisted with `sharedb-mongo`
+  // FIXME: This doesn't handle nested doc properties with dots like: {'_m.mtime': 12300}
+  function castToSnapshotQuery(query) {
```

Comment:
Alternate option to explore would be casting the document data into the same format that sharedb-mongo uses rather than casting the query. Advantage here would be that we'd be more sure it would work like sharedb-mongo for all types of queries. Should possible to make either work, however.